### ISSUE TITLE
Container stops on github environment: Error response from deamon: Container is not running

### ISSUE THREAD

**Issue by danielGz** [testcontainers/testcontainers-java#4265]
Hi, we're facing some issues with pipelines ran in github, we're using GenericContainer to execute some commands on a running container but after the container id is echoed and the command is attempted we get this:
```
Error response from daemon: Container <> is not running
  ] 
  Expecting:
   <1>
  to be equal to:
   <0>
  but was not.
```

Please take a look at our pipeline: https://github.com/spinnaker/clouddriver/pull/5414/checks?check_run_id=2967751843

The pipeline was ran with 1.15.1 but increasing to 1.15.3 didn't solve the problem. This only happens on github, running locally has no problems.

**Comment by kiview** [comment]
Can you please [configure logging](https://www.testcontainers.org/supported_docker_environment/logging_config/), so that we see Testcontainers logs? Else it is hard to understand what is happening. Also, a link to the implementation of the test would be helpful.

In addition, the error you posted does not come from Testcontainers, but from your test:
```
org.opentest4j.AssertionFailedError: [Running [sh, -c, ${KUBECTL_PATH} --kubeconfig=/home/runner/work/clouddriver/clouddriver/clouddriver-kubernetes/build/kubeconfigs/kubecfg-mystifying_shamir.yml create ns account1-testns00]
```

**Comment by danielGz** [comment]
agreed, the error comes from the test because that test relies on the running container, it all looks fine but when trying to exec into the container the container is not found

**Comment by kiview** [comment]
See my comment about logging and test implementation. Without this information, there is not much we can do.

**Comment by danielGz** [comment]
https://github.com/spinnaker/clouddriver/pull/5417/checks?check_run_id=3010259683
This run has the logging enabled as per this link: [config](https://www.testcontainers.org/supported_docker_environment/logging_config/), here are the tests: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/InfrastructureIT.java and this is the class responsible for starting the container: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/containers/KubernetesCluster.java

**Comment by bsideup** [comment]
@danielGz as you're running a k8s cluster in a container, can it be that the container is getting killed due to OOM?

**Comment by danielGz** [comment]
it could be, but what's weird is that this started happening out of nowhere, this repo has been iterated so many times and only the past week we started facing the issue, I was thinking that something with github changed but I have been unable to find reports similar to this.

**Comment by kiview** [comment]
@danielGz  Can you please increase the log level for Testcontainers to debug?

@bsideup Wouldn't we see an exception if the container is OOM killed?

**Comment by danielGz** [comment]
I increased the log level as requested, but I also increased the k3s ver of the docker image and the error is no longer, we had another error but it's a test case related issue :) https://github.com/spinnaker/clouddriver/pull/5419/checks?check_run_id=3011407918
